idaes/ui/flowsheet.py

```diff
@@ -380,6 +381,24 @@ def _construct_output_json(self):
         self._construct_jointjs_json()
 
     def _construct_model_json(self):
+        from idaes.core.util.tables import create_stream_table_dataframe # deferred to avoid circular import
+
+        # Get the stream table and add it to the model json
+        self._stream_table_df = create_stream_table_dataframe(self.arcs)
+        
+        # Change the index of the pandas dataframe to not be the variables
+        self._stream_table_df = self._stream_table_df.reset_index().rename(columns={"index": "Variable"})
+        self._stream_table_df = self._stream_table_df.reset_index().rename(columns={"index": ""})
+
+        # Parse the names of the variables to get rid of flow_mol_phase_comp
+        self._stream_table_df['Variable'] = self._stream_table_df['Variable'].map(lambda x: x.replace("flow_mol_phase_comp ", "") if "flow_mol_phase_comp" in x else x)
```

Comment:
Not sure if this would be useful to consider, but pandas has some utility functions to deal with text operations on `Series`/columns that could come in handy for this type of things: [Working with text data — pandas 1.2.0 documentation](https://pandas.pydata.org/pandas-docs/stable/user_guide/text.html#string-methods)

Reply:
I don't think that `df.columns.str.replace()` will work for this case as I have to iterate over rows not columns. I could use `apply` instead of a `map` but the internal lambda function would be the same.

Reply:
Rather than `DataFrame.columns.str.replace()`, I had `Series.str.replace()` in mind; this uses the `Series.str` methods, which implement (a subset of) common string operations in a vectorized way (row-wise).

If I understand the use case here, this could end up looking something like this:

```py
df["Variable"] = (
	df["Variable"]
	.str.replace("flow_mol_phase_comp", "")
	.str.rstrip()
)
```

Sorry for the confusion (and I guess that pandas doesn't really make our life easier in this regard...)

Reply:
Ah that makes more sense. Your code worked. Thanks!